### PR TITLE
Re-adding feedback logic after partner ends session ( after the file …

### PR DESCRIPTION
**Pull request by keerthanamuniraj**
…restructure)

Links
-----
- Task: _link to Notion task : https://www.notion.so/upchieve/6abe4da08ab045eea31efb13148448d6?v=1d1d833174ef402195a1d75586a1356d&p=44b7e1007e4145d985d37c353e64e029

Description
-----------
Known issue : Currently when a student ends a session, then the session of the volunteer is not ended automatically. This results in the feedback URL not showing up. This is true when a volunteer ends the session, the student's session is still active. The student currently has to go back to "Home" and then choose "Rejoin your coaching session". At this point, the web realizes the session has ended. 

I took out some time to fix this but it is a little tricky and needs synchronization between student and volunteer sessions.
    This PR attempts to put in a temporary fix where the user will see the user sees the feedback form even if they're rejoining a session that has ended.

Link to the old PR : https://github.com/UPchieve/web/pull/177
This PR became outdated with the file restructure. Hence recreated a branch to start over. ( The old branch had a lot of conflicts, made sense creating another PR) 
 
Developer self-review checklist
-------------------------------
- [x] Task's requirements have been fully addressed
- [x] Potentially confusing code has been explained with comments
- [x] Posted a link to this PR on the Notion task
- [x] No warnings or errors have been introduced; all known error cases have been handled
- [x] Any appropriate documentation (within the code, README.md, docs, etc) has been updated
- [x] There are no new spelling/grammar mistakes in the UI, code, or documentation
- [x] Branch has been deployed to staging, and all edge cases have been manually tested (on multiple browsers)
- [x] All new code complies with our ESLint standards

### CODE REVIEW

File: src/views/DashboardView/index.vue [hunk 1]
```diff
@@ -182,9 +183,19 @@ export default {
         console.log(`rejoining session: ${path}`)
       } else {
         console.log(`session terminated`)
-        this.$router.push('/')
+        //this.$router.push('/')
+        const user = UserService.getUser()
+        if (!user.isVolunteer) {
```
Comment (keerthanamuniraj) [inline src/views/DashboardView/index.vue]: Necessary to add two ifs back to back as both these conditions ( for student and volunteer ) have to be checked explicitly.

File: src/views/DashboardView/index.vue [hunk 2]
```diff
@@ -182,9 +183,19 @@ export default {
...
+          const url = '/feedback/null/null/null/student/null/null'
+          router.replace(url)
+        }
+        if (user.isVolunteer) {
+          const url = '/feedback/null/null/null/volunteer/null/null'
+          router.replace(url)
+        }
```
Comment (keerthanamuniraj) [inline src/views/DashboardView/index.vue]: Necessary to add two ifs back to back as both these conditions ( for student and volunteer ) have to be checked explicitly.

**Comment by keerthanamuniraj** [comment]
Resolved conflicts and retested the code.

File: src/views/DashboardView/index.vue (hunk 2 above)
Comment (dmnisson) [reply src/views/DashboardView/index.vue]: The way our database is set up right now, a user is _either_ a student _or_ a volunteer, and cannot be both or neither. Is there a situation where both of these blocks will be executed? If not, then we don't need to have two ifs back to back.

Comment (dmnisson) [reply src/views/DashboardView/index.vue]: The way it's written here is still good, though.

Comment (keerthanamuniraj) [reply src/views/DashboardView/index.vue]: Yeah, fair enough. The solution works with if and else. Made the change. Thanks.

**Pull request closed without merging**